core/api/common_api.hpp

```diff
@@ -43,9 +66,66 @@ namespace fc::api {
   template <typename A, typename F>
   void visitCommon(A &&a, const F &f) {
     f(a.AuthNew);
+    f(a.AuthVerify);
     f(a.NetAddrsListen);
     f(a.NetConnect);
     f(a.NetPeers);
     f(a.Version);
   }
+
+  inline void makeCommonApi(
+      const std::shared_ptr<CommonApi> &api,
+      const std::shared_ptr<ApiAlgorithm> &secret_algorithm,
+      const common::Logger &logger) {
+    api->AuthNew = [=](auto perms) -> outcome::result<Buffer> {
+      std::error_code ec;
+      auto token =
+          jwt::create()
+              .set_type(kTokenType)
+              .set_payload_claim(kPermissionKey,
+                                 jwt::claim(perms.begin(), perms.end()))
+              .sign(*secret_algorithm, ec);
+
+      if (ec) {
+        logger->error("Error when sign token: {}", ec.message());
+        return ERROR_TEXT("API ERROR");
+      }
+
+      return Buffer{common::span::cbytes(token)};
+    };
+    api->AuthVerify =
+        [=](auto token) -> outcome::result<std::vector<Permission>> {
+      static auto decode = [logger](const std::string &token)
+          -> outcome::result<jwt::decoded_jwt<jwt::picojson_traits>> {
+        try {
+          return jwt::decode(token);
+        } catch (const std::exception &e) {
+          logger->error("Error when decode token: {}", e.what());
+          return ERROR_TEXT("API ERROR");
+        }
+      };
+      static auto verifier = jwt::verify()
+                                 .with_type(kTokenType)
+                                 .allow_algorithm(*secret_algorithm);
+
+      OUTCOME_TRY(decoded_jwt, decode(token));
```

Comment:
```suggestion
      OUTCOME_TRY(decoded_jwt, outcomeCatch([&] { return jwt::decode(token); }));
```
jwt throws `std::system_error`, like out json and cbor